src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionIT.java

```diff
@@ -1,62 +1,231 @@
+          + "}";
 
-    OpenChannelRequest channelRequest =
-        OpenChannelRequest.builder(testChannelName)
-            .setDBName("TESTDB")
-            .setSchemaName("TESTSCHEMA")
-            .setTableName(tableName)
-            .setOnErrorOption(OpenChannelRequest.OnErrorOption.CONTINUE)
+  @BeforeEach
+  public void setUp() {
+    tableName = TestUtils.randomTableName();
+    topic = tableName;
+    topicPartition = new TopicPartition(topic, PARTITION);
+    Map<String, String> config = getConfForStreaming();
+    SnowflakeSinkConnectorConfig.setDefaultValues(config);
+    config.put(ICEBERG_ENABLED, "TRUE");
+    config.put(ENABLE_SCHEMATIZATION_CONFIG, "TRUE");
+
+    createIcebergTable(tableName);
+    enableSchemaEvolution(tableName);
+
+    // only insert fist topic to topicTable
+    Map<String, String> topic2Table = new HashMap<>();
+    topic2Table.put(topic, tableName);
+
+    service =
+        SnowflakeSinkServiceFactory.builder(conn, IngestionMethodConfig.SNOWPIPE_STREAMING, config)
+            .setRecordNumber(1)
+            .setErrorReporter(new InMemoryKafkaRecordErrorReporter())
+            .setSinkTaskContext(new InMemorySinkTaskContext(Collections.singleton(topicPartition)))
+            .setTopic2TableMap(topic2Table)
+            .addTask(tableName, topicPartition)
             .build();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    if (service != null) {
+      service.closeAll();
+    }
+    dropIcebergTable(tableName);
+  }
+
+  @ParameterizedTest()
+  @MethodSource("prepareData")
+  @Disabled
+  void shouldEvolveSchemaAndInsertRecords(
+      String description, String message, DescribeTableRow[] expectedSchema, boolean withSchema)
+      throws Exception {
+    // start off with just one column
+    List<DescribeTableRow> rows = describeTable(tableName);
+    assertThat(rows.size()).isEqualTo(1);
+    assertThat(rows.get(0).getColumn()).isEqualTo(Utils.TABLE_COLUMN_METADATA);
+
+    SinkRecord record = createKafkaRecord(message, 0, withSchema);
+    service.insert(Collections.singletonList(record));
+    TestUtils.assertWithRetry(() -> service.getOffset(topicPartition) == -1, 20, 5);
+    rows = describeTable(tableName);
+    assertThat(rows.size()).isEqualTo(9);
+
+    // don't check metadata column schema, we have different tests for that
+    rows =
+        rows.stream()
+            .filter(r -> !r.getColumn().equals(Utils.TABLE_COLUMN_METADATA))
+            .collect(Collectors.toList());
+
+    assertThat(rows).containsExactlyInAnyOrder(expectedSchema);
+
+    // resend and store same record without any issues now
+    service.insert(Collections.singletonList(record));
+    TestUtils.assertWithRetry(() -> service.getOffset(topicPartition) == 1, 20, 5);
+
+    // and another record with same schema
+    service.insert(Collections.singletonList(createKafkaRecord(message, 1, withSchema)));
+    TestUtils.assertWithRetry(() -> service.getOffset(topicPartition) == 2, 20, 5);
```

Comment:
nit: having overloaded `assertWithRetry` without interval details might be a good option